alternative-flows.md

```diff
@@ -0,0 +1,147 @@
+impersonation), and present the necessary identity token to the authorization service asserting the impersonated WebID.
+
+#### Example:
+- Suppose an application is provided by a company's IT team to deliver invoices to Solid Pods. 
+The different business units of the organisation have their own WebID and are authorized out-of-band to deliver an invoice to their customer's Pod.
+This central invoicing service should be able to assume the identity of each of these business units in order to use their pre-existing authorizations.
+
+### (Shared) Input-Constrained devices
+Some applications may be tied to input-constrained devices where a typical redirection flow cannot be performed. Also, these devices may or
+may not be shared by multiple end-users. While sometimes a service account may be most appropriate in this context, other times it might be 
+necessary for these devices to assume several distinct identities (WebIDs).
+
+#### Example:
+- Suppose a smart picture frame wants to access your own photos as well as pictures that your friends have shared with you. Therefore it will
+need to assume your WebID in its interactions with these resources shared by third parties. The oAuth 2.0 framework provides the [Device Code
+flow](https://oauth.net/2/grant-types/device-code/) for this exact purpose, however this flow would not yield the identity token needed to 
+integrate with the existing Solid-OIDC flow.
+
+## Proposed solutions
+The following section proposes a number of solutions that could be used to integrate these flows within the existing Solid-OIDC specification.
+
+### 1. OAuth 2.0 Token Exchange
+The [OAuth 2.0 Token Exchange specification](https://datatracker.ietf.org/doc/html/rfc8693) defines a mechanism for applying the concept of
+a Security Token Service to OAuth 2.0. A Security Token Service (STS) is a service capable of validating security tokens provided to it and
+issuing new security tokens in response, which enables clients to obtain appropriate access credentials for resources in heterogeneous
+environments or across security domains. The mechanics of this might be applicable if we want to allow clients to exchange an OAuth 2.0
+Access Token for a Solid-OIDC Identity Token.
+
+#### Proposed use
+Suppose a client application has already obtained an OAuth 2.0 access token through a supported flow, like Device Code Grant or
+Client Credentials Grant, and needs to exchange this for a OIDC ID token. 
+
+If the OIDC IdP, referenced by the WebID that is to be assumed by the application, supports our proposed token exchange flow to obtain an 
+ID token a sequence of requests similar to what is shown below could be envisioned:
+
+1. **Preconditions:**
+    - An OAuth Access token has been obtained by the client.
+    - The OIDC Issuer's Token Exchange endpoint is able to determine for some Access Token which WebID it is authorized to obtain an ID 
+    Token for.
+2. Token exchange request is made by the client:
```

Comment:
Presumably, the client would also indicate `scope=solid`

Reply:
Agreed